pure.zsh

```diff
@@ -143,11 +143,10 @@ prompt_pure_preprompt_render() {
 	# Add Git branch and dirty status info.
 	typeset -gA prompt_pure_vcs_info
 	if [[ -n $prompt_pure_vcs_info[branch] ]]; then
-		local branch="%F{$git_color}"'${prompt_pure_vcs_info[branch]}'
+		preprompt_parts+=("%F{$git_color}"'${prompt_pure_vcs_info[branch]}'"%F{$git_dirty_color}"'${prompt_pure_git_dirty}%f')
 		if [[ -n $prompt_pure_vcs_info[action] ]]; then
-			branch+="|%F{$prompt_pure_colors[git:action]}"'$prompt_pure_vcs_info[action]'"%F{$git_color}"
+			preprompt_parts+=($'\e[2;90m(\e[0m'"%F{$prompt_pure_colors[git:action]}"'$prompt_pure_vcs_info[action]%f'$'\e[2;90m)\e[0m')
```

Comment:
What's the purpose of these escape codes? If we're keeping them, a comment on why they're there would be good.

If they're related to changing color (e.g. dimming?) then I think it's best to remove them as they would hinder the user from selecting a brighter color and instead change the default from `yellow` to a dimmer yellow.

Reply:
If you are going to keep these escape codes, you need to wrap them in `%{...%}`. Otherwise prompt will break if it wraps around to the next line.

Reply:
Doh. Now I realized it's the parens 😅. Could we use ZSHs color formatting via `%F{x}%f` as it's a bit more readable?

I imagine the hardcoded color could be problematic with certain terminal backgrounds (at least on one that matches the color used in the parens).

Reply:
> If you are going to keep these escape codes, you need to wrap them in `%{...%}`. Otherwise prompt will break if it wraps around to the next line.

That's a good point, `%F` should take care of that as well, right?

Reply:
There are two escape codes: one is foreground color (the same as `%F{60}`), the other is dimmed (no percent-escape for it in Zsh).

Reply:
So like this?

```suggestion
			preprompt_parts+=($'%{\e[2m%}%F{7}(\e[0m'"%F{$prompt_pure_colors[git:action]}"'$prompt_pure_vcs_info[action]'$'%{\e[2m%}%F{7})\e[0m')
```

Reply:
I haven't tried it but it looks correct to me.

By the way, using dim style is a *very* unusual choice. Several popular terminals don't support it. Up to this point Pure mostly stayed within the well-trodden path of portable terminal features.

Disclaimer: If you commit this, I'll be obliged to make the same change in Pure style of Powerlevel10k (there is an option in Powerlevel10k to show the same prompt as Pure). I really don't want a stream of bug reports due to the use of obscure terminal features. If you can, please use portable code.